backend/layers/business/business.py

```diff
@@ -578,6 +579,34 @@ def get_dataset_version_from_canonical(self, dataset_id: DatasetId) -> Optional[
                     unpublished_dataset = dataset
             return unpublished_dataset
 
+    def get_prior_published_versions_for_dataset(self, dataset_id: DatasetId) -> List[PublishedDatasetVersion]:
+        """
+        Given a canonical dataset id, return all its DatasetVersions that have been part of published CollectionVersions
+        """
+        dataset_version = self.database_provider.get_dataset_mapped_version(dataset_id)
+        if not dataset_version:
+            return []
+        collection_versions = self.database_provider.get_all_versions_for_collection(dataset_version.collection_id)
+        published_version_history = []
+        found_version_ids = set()
+        # sort to ensure we always find earliest instance of a dataset version first when iterating
+        collection_versions = sorted(collection_versions, key=lambda cv: (cv.published_at is None, cv.published_at))
```

Comment:
how does the sorting for a tuple key work?

Reply:
In general, tuple keys will prioritize sorting on the first entry and use the second entry as the tiebreaker sort

In this scenario that means it'll always sort unpublished versions to the end of the list (where published_at is None is True), then sort by published_at among the published versions

Reply:
for added context: the reason Trues go to the end is because its actually sorting on False vs. True alphabetically

Reply:
also necessary to sort while handling the None values, otherwise you'll get a Type Error on the sort

https://stackoverflow.com/questions/48234072/how-to-sort-a-list-and-handle-none-values-properly

Reply:
>for added context: the reason Trues go to the end is because its actually sorting on False vs. True alphabetically

this is very... coincidental / orthogonal to the actual entity. Lol. Can you add a comment explaining this if you're going to use this tactic? I mean I'm not wholly opposed, but in the absence of an explicit lambda that shows the logic, it would be good to have a comment.

Reply:
tbf we're filtering out the Nones anyway so whether it gets sorted to the end or the start is actually irrelevant (the False vs. True stuff is more a fun fact/quirk of the implementation). The real purpose is to allow a list with a NoneType to be sorted by datetime. The other option is to iterate through the list once to filter out unpublished versions before sorting, which is maybe more clear but less efficient since we add an extra loop. 

tl;dr: I'll add a comment saying this is done to avoid a TypeError on sorting, the implementation details aren't super relevant